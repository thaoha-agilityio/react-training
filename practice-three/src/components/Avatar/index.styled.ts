import styled from "styled-components";

type StyledAvatarProps = {
  width?: number;
  height?: number;
  borderRadius: number;
}

const AvatarStyled = styled.img<StyledAvatarProps>`
  ${(props) => `width:${props.width}px`};
  ${(props) => `height:${props.height}px`};
  ${(props) => `border-radius:${props.borderRadius}px`};
`;

export { AvatarStyled };
