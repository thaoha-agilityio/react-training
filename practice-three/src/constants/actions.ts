export const KEY_NAME_ESC = 27;
