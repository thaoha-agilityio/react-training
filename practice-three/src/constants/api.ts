const API_BASE_URL = process.env.VITE_BASE_URL;

const API_PATH = {
  BOOKS: "/books",
  CATEGORIES: "/categories",
};

export { API_PATH, API_BASE_URL };
