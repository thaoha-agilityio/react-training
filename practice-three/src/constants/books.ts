export const THUMBNAIL = `https://media.gq.com/photos/5ad64204c8be07604e8b5f2f/4:3/w_1776,h_1332,c_limit/21-books-GQ-April-2018-041718-3x2.jpg`;
