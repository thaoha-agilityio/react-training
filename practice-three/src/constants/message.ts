export const NOTICE_MESSAGE = "No records to display";
export const ERROR_MESSAGE = "URL is not valid";
