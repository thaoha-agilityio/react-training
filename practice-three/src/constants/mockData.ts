export const THUMBNAIL = `https://media.gq.com/photos/5ad64204c8be07604e8b5f2f/4:3/w_1776,h_1332,c_limit/21-books-GQ-April-2018-041718-3x2.jpg`;

export const book = {
  name: "Don Quijote de la Mancha",
  avatar:
    "https://image.cdn0.buscalibre.com/5c7482160bf0b5fd0c8b4567.__RSF640x640__.jpg",
  author: "libsa",
  description: "abc",
  published: 2016,
  publishers: "Printer",
  id: "a",
};

export const category = {
  id: "1",
  name: "romance",
  total: 12,
  bgColor: "#0DA8FF",
};

export const categories = [
  {
    name: "Adventure",
    id: "1",
    total: 123,
    bgColor: "#F388B5",
  },
  {
    name: "Romance",
    id: "2",
    total: 123,
    bgColor: "#94AAE4",
  },
];

export const books = {
  data: [
    {
      name: "Don Quijote de la Mancha",
      avatar:
        "http://betweenthelines.in/wp-content/uploads/2012/09/Life-is-what-you-make-it-11.jpg",
      author: "Thomas Harris",
      description:
        "he serial killer nicknamed 'Buffalo Bill' has been capturing and starving women, then murdering and skinning them. FBI rookie Clarice Starling is assigned to solicit help from imprisoned psychopath Dr Hannibal 'the Cannibal' Lecter, whose insight into the depraved minds of serial killers is second to none",
      published: 2012,
      publishers: "Paperback",
      categoryId: "1",
      id: "1",
    },
    {
      name: "Silence Of The Lambs",
      avatar:
        "https://kbimages1-a.akamaihd.net/bc050d09-4422-4be0-bda6-49941b0cae5e/1200/1200/False/silence-of-the-lambs.jpg",
      author: "Thomas Harris",
      description:
        "The serial killer nicknamed 'Buffalo Bill' has been capturing and starving women, then murdering and skinning them. FBI rookie Clarice Starling is assigned to solicit help from imprisoned psychopath Dr Hannibal 'the Cannibal' Lecter, whose insight into the depraved minds of serial killers is second to none.",
      published: 1980,
      publishers: "Hannibal Lecter",
      categoryId: "1",
      id: "2",
    },
    {
      name: "Angels And Demons",
      avatar: "https://pictures.abebooks.com/isbn/9780552159708-fr.jpg",
      description: "description 4",
      author: "Dan Brown",
      published: 42,
      publishers: "Hannibal Lecter",
      categoryId: "3",
      id: "3",
    },
  ],
};
