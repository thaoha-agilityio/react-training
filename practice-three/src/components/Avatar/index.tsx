import { AvatarStyled } from "./index.styled";

import { THUMBNAIL } from "../../constants/books";

interface IProps {
  url: string;
  alt: string;
  width: number;
  height: number;
  borderRadius: number;
}

const Avatar = ({
  width,
  height,
  alt,
  url = THUMBNAIL,
  borderRadius,
}: IProps): React.ReactElement => (
  <AvatarStyled
    src={url}
    alt={alt}
    borderRadius={borderRadius}
    width={width}
    height={height}
  />
);

export default Avatar;
