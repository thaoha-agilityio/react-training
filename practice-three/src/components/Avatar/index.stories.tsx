import { ComponentStory, ComponentMeta } from "@storybook/react";

import { THUMBNAIL } from "@/constants/mockData";

import Avatar from ".";

export default {
  title: "Avatar",
  component: Avatar,
  argTypes: {
    width: {
      defaultValue: 200,
      control: { type: "number" },
    },
    height: {
      defaultValue: 200,
      control: { type: "number" },
    },
    url: {
      defaultValue: THUMBNAIL,
    },
    alt: {
      defaultValue: "avatar",
    },
    borderRadius: {
      defaultValue: 5,
    },
  },
  parameters: {},
} as ComponentMeta<typeof Avatar>;

const Template: ComponentStory<typeof Avatar> = (args) => <Avatar {...args} />;

export const Default = Template.bind({});

export const ImageAvatar = Template.bind({});
ImageAvatar.args = {
  ...ImageAvatar.args,
  width: 353,
  height: 144,
};
