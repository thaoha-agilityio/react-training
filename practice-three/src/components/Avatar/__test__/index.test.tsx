import { render, screen } from '@testing-library/react';

import { THUMBNAIL } from '../../../constants/mockData';

import Avatar from '..';

const mockProps = {
  width: 201,
  height: 202,
  borderRadius: 5,
  alt: 'avatar',
  url: THUMBNAIL,
};

describe('Avatar render', () => {
  test('Ensure image avatar render correctly', () => {
    render(<Avatar {...mockProps} />);
    const imageAvatar = screen.getByRole('img');

    expect(imageAvatar.getAttribute('src')).toBe(THUMBNAIL);
    expect(imageAvatar.getAttribute('alt')).toBe('avatar');
  });
});
