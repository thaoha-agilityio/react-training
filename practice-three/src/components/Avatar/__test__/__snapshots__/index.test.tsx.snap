// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar render Ensure image avatar render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="avatar"
        class="sc-bcXHqe kHhOGL"
        height="202"
        src="https://media.gq.com/photos/5ad64204c8be07604e8b5f2f/4:3/w_1776,h_1332,c_limit/21-books-GQ-April-2018-041718-3x2.jpg"
        width="201"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="avatar"
      class="sc-bcXHqe kHhOGL"
      height="202"
      src="https://media.gq.com/photos/5ad64204c8be07604e8b5f2f/4:3/w_1776,h_1332,c_limit/21-books-GQ-April-2018-041718-3x2.jpg"
      width="201"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
